src/ledger/submit.ts

```diff
@@ -0,0 +1,55 @@
+import { encode } from 'ripple-binary-codec'
+
+import type { Client, SubmitRequest, SubmitResponse, Wallet } from '..'
+import { Transaction } from '../models/transactions'
+import { sign } from '../wallet/signer'
+
+import autofill from './autofill'
+
+/**
+ * Submits an unsigned transaction.
+ * Steps performed on a transaction:
+ *    1. Autofill.
+ *    2. Sign & Encode.
+ *    3. Submit.
+ *
+ * @param client - A Client.
+ * @param wallet - A Wallet to sign a transaction.
+ * @param transaction - A transaction to autofill, sign & encode, and submit.
+ * @returns A promise that contains SubmitResponse.
+ * @throws RippledError if submit request fails.
+ */
+async function submitTransaction(
+  client: Client,
+  wallet: Wallet,
+  transaction: Transaction,
+): Promise<SubmitResponse> {
+  const tx: Transaction = await autofill(client, transaction)
+  const signedTxEncoded: string = sign(wallet, tx)
+  return submitSignedTransaction(client, signedTxEncoded)
+}
+
+/**
+ * Encodes and submits a signed transaction.
+ *
+ * @param client - A Client.
+ * @param signedTransaction - A signed transaction to encode (if not already) and submit.
+ * @returns A promise that contains SubmitResponse.
+ * @throws RippledError if submit request fails.
+ */
+async function submitSignedTransaction(
+  client: Client,
+  signedTransaction: Transaction | string,
+): Promise<SubmitResponse> {
+  const signedTxEncoded =
+    typeof signedTransaction === 'string'
+      ? signedTransaction
+      : encode(signedTransaction)
+  const request: SubmitRequest = {
```

Comment:
```suggestion
  const request = {
```

Reply:
When I remove `: SubmitRequest`, TS complains because `client.request()` is overloaded with multiple request types. We need to keep this change as a way to tell TS we're calling the `SubmitRequest` version of `client.request()`

Reply:
It should handle the overload automatically - let me take a look

Reply:
Hm that's strange - ok, leave it in for now